.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,190 @@
+# This is a workflow for opticlust ci-cd
+
+name: CI/CD
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the "main and develop" branch
+  push:
+    branches: [ "main", "develop" ]
+  pull_request:
+    branches: [ "main", "develop" ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  Ruff_linter:
+    # Set up operating system
+    runs-on: ubuntu-latest
+    
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+
+    - name: Set up Python 3.12
+      uses: actions/setup-python@v2

```

Comment:
I'll report the latest actions here (for fun). No need to change any unless you feel like it.

- `actions/setup-python@v5.3.0` https://github.com/actions/setup-python/releases
- `actions/checkout@v4.2.2` https://github.com/actions/checkout/releases
- `actions/upload-artifact@v4.4.3` https://github.com/actions/upload-artifact/releases
- `actions/download-artifact@v4.1.8` https://github.com/actions/download-artifact/releases

Reply:
there might even be a way to automatically update action version: https://stackoverflow.com/questions/78380016/how-to-automatically-update-versions-of-github-actions-with-renovate